### DISCUSSION TITLE
v3.0.2

### DISCUSSION THREAD

**Discussion by ghenry** [SentryPeer/SentryPeer#92] [Announcements]
### Fixes
- Check `user_agent_header->hvalue` is not NULL as you can get a blank user-agent value in malformed SIP packets

<hr /><em>This discussion was created from the release <a href='https://github.com/SentryPeer/SentryPeer/releases/tag/v3.0.2'>v3.0.2</a>.</em>

**Comment by zandeez** [comment]
Hi Gavin,

I've also come across this this week, but 3CX sends initial REGISTER with no User-agent header. Caused me a few headaches :)

My workaround was to check the request URI was a FQDN rather than an IP address.

**Reply by ghenry** [reply]
I need to check if it's legit as per the SIP spec, but the crap that comes in means you need to be vigilant 😎

**Reply by zandeez** [reply]
Yeah I know what you mean, 3CX forums are full of "my trunk provider needs User-agent" and 3CX are holding firm. They're not winning many friends this year 🤣

**Reply by zandeez** [reply]
So, I know of a provider that you may have heard of that, since changing to a new platform earlier this month, no longer sends User-agent headers in INVITEs.

**Reply by ghenry** [reply]
That's nice of them!
